src/Integration.Vsix/SonarLintTagger/README.md

```diff
@@ -82,3 +82,47 @@ Error list integration:
 - `SinkManager` gets updates through factories.
 
 
+
+Tracking document edits:
+------------------------
+See MS docs "Inside the editor": https://docs.microsoft.com/en-us/visualstudio/extensibility/inside-the-editor?view=vs-2019
+
+A text file opened in the VS Editor is represented as an ITextBuffer. The buffer can be thought of as series of
+immutable ITextSnapshots.  A region of text in a ITextSnapshot is represented by a SnapshotSpan.
+Each time the buffer is edited, a new ITextSnapshot is created. The buffer tracks the changes between snapshots,
+making it possible to translate from a SnapshotSpan in the previous snapshot to the corresponding SnapshotSpan
+in the new ITextSnapshot e.g.
+* lines are added at the beginning of the buffer -> all old spans can be translate to spans in the new snapshot.
+* lines are added at the end of the buffer -> all old spans can be translated to snaps in the new snapshot.
+   The old spans still need to be translated, even though the positions of the span haven't changed.
+* a line is deleted from the buffer -> any spans that included positions in the deleted line cannot be translated
+   to spans in the new snapshot. All other old spans can be translated to new spans.
+ 
+A SnapshotSpan can be translated across multiple snapshot versions e.g. from snapshot v1 to snapshot v3.
+
+Implementation note: the VS Editor also supports "tracking spans" that work across snapshots. The Roslyn tagger
```

Comment:
Hm I'm not sure what you mean by tracking spans. Do you think it's worth elaborating, or is it irrelevant?

Reply:
I've updated the comment. Tracking spans automatically adjust as the snapshot changes so there is no need to manually translate SnapshotSpans between snapshots.